File: 백수/algorithm/문자열/숫자문자열과영단어.py
def solution(s):
    answer = ''
    dic={'zero':'0','one':'1','two':'2','three':'3','four':'4','five':'5','six':'6','seven':'7','eight':'8','nine':'9'}
    tmp_s=''
    for c in s:
        tmp_s+=c
        if c.isdigit():
            answer+=c
            tmp_s=''
            continue
            
        if tmp_s in dic:
            answer+=dic[tmp_s]
            tmp_s=''
    
    return int(answer)

File: 백수/algorithm/문자열/숫자 문자열과 영단어.java
import java.util.*;
class Solution {
    public int solution(String s) {
        String answer = "";
        Map<String,String> dic=new HashMap<String,String>();
        String[] wordArray=s.split("");
        String stack="";
        char c;

        dic.put("zero","0");
        dic.put("one","1");
        dic.put("two","2");
        dic.put("three","3");
        dic.put("four","4");
        dic.put("five","5");
        dic.put("six","6");
        dic.put("seven","7");
        dic.put("eight","8");
        dic.put("nine","9");
        Set<String> keys=dic.keySet();
        
        for(int i=0;i<s.length();i++){
            c=s.charAt(i);
            
            if(Character.isDigit(c)==true){
                answer+=c;
            }else{
                stack+=c;
                if(dic.containsKey(stack)){
                    answer+=dic.get(stack);
                    stack="";
                }
            }
        }
//         for(String c:wordArray){
//             if(Character.isDigit(c)==true){
//                 System.out.println(c);
//             }
            
//         }


        return Integer.valueOf(answer);
    }
}